core/src/serve_repair.rs

```diff
@@ -719,6 +778,75 @@ impl ServeRepair {
         Some(res)
     }
 
+    fn broadcast_only_coding_shreds_enabled(
+        bank_forks: Option<&Arc<RwLock<BankForks>>>,
+        slot: Slot,
+    ) -> bool {
+        if bank_forks.is_none() {
+            // enable for testing
+            return true;
+        }
+        let bank_forks = bank_forks.unwrap();
+
+        let working_bank = bank_forks.read().unwrap().working_bank();
+        let feature_slot = working_bank
+            .feature_set
+            .activated_slot(&feature_set::broadcast_only_coding_shreds::id());
+        if let Some(feature_slot) = feature_slot {
+            let epoch_schedule = working_bank.epoch_schedule();
+            let feature_epoch = epoch_schedule.get_epoch(feature_slot);
+            let slot_epoch = epoch_schedule.get_epoch(slot);
+            feature_epoch < slot_epoch
+        } else {
+            false
+        }
+    }
+
+    fn run_coding_request(
+        recycler: &PacketBatchRecycler,
+        from: &ContactInfo,
+        from_addr: &SocketAddr,
+        blockstore: Option<&Arc<Blockstore>>,
+        bank_forks: Option<&Arc<RwLock<BankForks>>>,
+        my_id: &Pubkey,
+        slot: Slot,
+        shred_index: u64,
+        nonce: Nonce,
+    ) -> Option<PacketBatch> {
+        if Self::broadcast_only_coding_shreds_enabled(bank_forks, slot) {
```

Comment:
what is the motivation to add and check for this feature?
if someone has sent request for a coding shred, there is no harm to send him back the coding shred if we have it.